ntp-proto/src/packet/extensionfields.rs

```diff
@@ -375,6 +418,28 @@ impl<'a> ExtensionField<'a> {
         })
     }
 
+    #[cfg(feature = "ntpv5")]
+    fn decode_draft_identification(
+        message: &'a [u8],
+    ) -> Result<Self, ParsingError<std::convert::Infallible>> {
+        let msg = message[..].into();
+
+        let new_msg = match msg {
+            Cow::Borrowed(slice) => std::str::from_utf8(slice).ok().map(Cow::Borrowed),
+            Cow::Owned(vec) => String::from_utf8(vec).ok().map(Cow::Owned),
+        };
```

Comment:
the into will always choose `Borrowed` here (right?). So handling the full Cow confuses me